src/vue/vuex-modules/module.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * A base class for all data and UI modules. Its role is to define common features that are shared
```

Comment:
I am not completely sure yet, but UI modules might have a store. Let's say that if a popup UI  module has a specific message panel that a panel UI do not have then it would make sense for a popup UI module to integrate into the Vuex store exposing a `panelMessage` prop to it. Whoever may want to display a message in that specific panel would have to mutate that prop.

But I'm still not sure what's the best architecture here. What do you think?